src/automatic_differentiation.jl

```diff
@@ -308,21 +322,21 @@ julia> ∇f = gradient(norm, A)
 julia> ∇f, f = gradient(norm, A, :all);
 ```
 """
-function Base.gradient{F}(f::F, v::Union{SecondOrderTensor, Vec})
+function Base.gradient{F <: Function}(f::F, v::Union{SecondOrderTensor, Vec, Real})
```

Comment:
Har to do this `<: Function` because of an ambiguity with some function in `Calculus` otherwise... We be type piracying all over the place..